roles/openshift_health_checker/openshift_checks/__init__.py

```diff
@@ -19,15 +19,26 @@ class OpenShiftCheckException(Exception):
 
 @six.add_metaclass(ABCMeta)
 class OpenShiftCheck(object):
-    """A base class for defining checks for an OpenShift cluster environment."""
+    """
+    A base class for defining checks for an OpenShift cluster environment.
+
+    Expect optional params: method execute_module, dict task_vars, and string tmp.
+    execute_module is expected to have a signature compatible with _execute_module
+    from ansible plugins/action/__init__.py, e.g.:
+    def execute_module(module_name=None, module_args=None, tmp=None, task_vars=None, *args):
+    This is stored so that it can be invoked in subclasses via check.execute_module("name", args)
+    which provides the check's stored task_vars and tmp.
+    """
 
-    def __init__(self, execute_module=None, module_executor=None):
-        if execute_module is module_executor is None:
-            raise TypeError(
-                "__init__() takes either execute_module (recommended) "
-                "or module_executor (deprecated), none given")
-        self.execute_module = execute_module or module_executor
-        self.module_executor = self.execute_module
+    def __init__(self, execute_module=None, task_vars=None, tmp=None):
+        def _placeholder(*_):
+            raise RuntimeError(
+                self.__class__.__name__ +
+                " invoked execute_module without providing the method at initialization."
+            )
+        self._execute_module = execute_module or _placeholder
```

Comment:
Suggestion:

```python
def __init__(...):
    self._execute_module = execute_module

...

def execute_module(...):
    ...
    if self._execute_module is None:
        raise NotImplementedError("'execute_module' was not defined at instance initialization.")
    return self._execute_module(module_name, module_args, self.tmp, self.task_vars)
```

This would keep `__init__` clean and straightforward.